src/wp-includes/update.php

```diff
@@ -956,6 +956,52 @@ function wp_clean_update_cache() {
 	delete_site_transient( 'update_core' );
 }
 
+/**
+ * Deletes all contents of the temp-backup directory.
+ *
+ * @since 6.0.0
+ *
+ * @global WP_Filesystem_Base $wp_filesystem WordPress filesystem subclass.
+ */
+function wp_delete_all_temp_backups() {
+	/*
+	 * Check if there's a lock, or if currently performing an Ajax request,
+	 * in which case there's a chance we're doing an update.
+	 * Reschedule for an hour from now and exit early.
+	 */
+	if ( get_option( 'core_updater.lock' ) || get_option( 'auto_updater.lock' ) || wp_doing_ajax() ) {
+		wp_schedule_single_event( time() + HOUR_IN_SECONDS, 'wp_delete_temp_updater_backups' );
+		return;
+	}
+
+	add_action(
+		'shutdown',
+		/*
+		 * This action runs on shutdown to make sure there's no plugin updates currently running.
+		 * Using a closure in this case is OK since the action can be removed by removing the parent hook.
+		 * `remove_action( 'wp_delete_temp_updater_backups', 'wp_delete_all_temp_backups' );`
+		 */
+		function() {
```

Comment:
Can this closure be made a `static` one, or does this cause an issue?

Reply:
I have no idea.

Reply:
@costdev  I'm never used a static closure.  What benefits do they provide?

Reply:
@pbiron

> Anonymous functions may be declared statically. This prevents them from having the current class automatically bound to them. Objects may also not be bound to them at runtime.

See [PHP.net - Static Anonymous Functions](https://www.php.net/manual/en/functions.anonymous.php#functions.anonymous-functions.static).

See [when to use static closures](https://www.designcise.com/web/tutorial/what-are-static-anonymous-functions-in-php#when-to-use-static-closures).

In most situations, it's a micro-optimization, but it can also make a noticeable difference in others. It's often used in WP Core Tests, for example.